src/Avalonia.Visuals/Media/Rotate3DTransform.cs

```diff
@@ -0,0 +1,190 @@
+using System;
+using System.Numerics;
+using Avalonia.Animation.Animators;
+
+namespace Avalonia.Media;
+
+/// <summary>
+///  Non-Affine 3D transformation for rotating an visual around a definable axis
+/// </summary>
+public class Rotate3DTransform : Transform
+{
+    /// <summary>
+    /// Defines the <see cref="AngleX"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> AngleXProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(AngleX));
+
+    /// <summary>
+    /// Defines the <see cref="AngleY"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> AngleYProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(AngleY));
+
+    /// <summary>
+    /// Defines the <see cref="AngleZ"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> AngleZProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(AngleZ));
+
+
+    /// <summary>
+    /// Defines the <see cref="CenterX"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> CenterXProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(CenterX));
+
+
+    /// <summary>
+    /// Defines the <see cref="CenterY"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> CenterYProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(CenterY));
+
+
+    /// <summary>
+    /// Defines the <see cref="CenterZ"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> CenterZProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(CenterZ));
+
+    /// <summary>
+    /// Defines the <see cref="Depth"/> property.
+    /// </summary>
+    public static readonly StyledProperty<double> DepthProperty =
+        AvaloniaProperty.Register<Rotate3DTransform, double>(nameof(Depth));
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Rotate3DTransform"/> class.
+    /// </summary>
+    public Rotate3DTransform() { }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Rotate3DTransform"/> class.
+    /// </summary>
+    /// <param name="angleX">The rotation around the X-Axis</param>
+    /// <param name="angleY">The rotation around the Y-Axis</param>
+    /// <param name="angleZ">The rotation around the Z-Axis</param>
+    /// <param name="centerX">The origin of the X-Axis</param>
+    /// <param name="centerY">The origin of the Y-Axis</param>
+    /// <param name="centerZ">The origin of the Z-Axis</param>
+    public Rotate3DTransform(
+        double angleX,
+        double angleY,
+        double angleZ,
+        double centerX,
+        double centerY,
+        double centerZ) : this()
+    {
+        AngleX = angleX;
```

Comment:
To avoid multiple call RaiseChanged add flag isInitialized and call RaiseChanged  after all properties is sets.